src/features/terminal/runInTerminal.ts

```diff
@@ -29,10 +28,21 @@ export async function runInTerminal(
                 deferred.resolve();
             }
         });
+
+        const shouldSurroundWithQuotes =
```

Comment:
wrapping executable in double quotes after checking if there is white space.
Also checking they are not already wrapped in double quotes.